hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientReliableTopicProxy.java

```diff
@@ -197,6 +212,115 @@ public LocalTopicStats getLocalTopicStats() {
+                    .map(m -> new ReliableTopicMessage(toData(m), null))
+                    .collect(Collectors.toList());
+            switch (overloadPolicy) {
+                case ERROR:
+                    long sequenceId = ringbuffer.addAllAsync(messages, OverflowPolicy.FAIL).toCompletableFuture().get();
+                    if (sequenceId == -1) {
+                        throw new TopicOverloadException("Failed to publish messages: " + payload + " on topic:" + getName());
+                    }
+                    break;
+                case DISCARD_OLDEST:
+                    ringbuffer.addAllAsync(messages, OverflowPolicy.OVERWRITE).toCompletableFuture().get();
+                    break;
+                case DISCARD_NEWEST:
+                    ringbuffer.addAllAsync(messages, OverflowPolicy.FAIL).toCompletableFuture().get();
+                    break;
+                case BLOCK:
+                    addWithBackoff(messages);
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unknown overloadPolicy:" + overloadPolicy);
+            }
+        } catch (Exception e) {
+            throw (RuntimeException) peel(e, null,
+                    "Failed to publish messages: " + payload + " to topic:" + getName());
+        }
+    }
+
+    @Override
+    public CompletionStage<Void> publishAllAsync(@Nonnull Collection<? extends E> payload) {
+        checkNotNull(payload, NULL_MESSAGE_IS_NOT_ALLOWED);
+        checkNoNullInside(payload, NULL_MESSAGE_IS_NOT_ALLOWED);
+        InternalCompletableFuture<Void> returnFuture = new InternalCompletableFuture<>();
+
+        try {
+
+            List<ReliableTopicMessage> messages = payload.stream()
+                    .map(m -> new ReliableTopicMessage(toData(m), null))
+                    .collect(Collectors.toList());
+            switch (overloadPolicy) {
+                case ERROR:
+                    addAsyncOrFail(payload, returnFuture, messages);
+                    break;
+                case DISCARD_OLDEST:
+                    addAsync(messages, OverflowPolicy.OVERWRITE);
+                    break;
+                case DISCARD_NEWEST:
+                    addAsync(messages, OverflowPolicy.FAIL);
+                    break;
+                case BLOCK:
+                    addAsyncAndBlock(payload, returnFuture, messages, INITIAL_BACKOFF_MS);
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unknown overloadPolicy:" + overloadPolicy);
+            }
+        } catch (Exception e) {
+            throw (RuntimeException) peel(e, null,
+                    String.format("Failed to publish messages: %s on topic: %s", payload, getName()));
+        }
+        return returnFuture;
+    }
+
+    private void addAsyncAndBlock(@Nonnull Collection<? extends E> payload, InternalCompletableFuture<Void> returnFuture,
```

Comment:
This payload parameter is unused. It can be removed. Please check the member side proxy as well.

Reply:
Removed